server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -407,27 +377,36 @@ public String executor() {
                 return ThreadPool.Names.GENERIC;
             }
         };
-        try {
-            logger.info("Sending extension request type: " + REQUEST_EXTENSION_ACTION_NAME);
-            transportService.connectToExtensionNode(extension);
-            transportService.sendRequest(
-                extension,
-                REQUEST_EXTENSION_ACTION_NAME,
-                new InitializeExtensionRequest(transportService.getLocalNode(), extension),
-                initializeExtensionResponseHandler
-            );
-            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
-        } catch (CompletionException | ConnectTransportException e) {
-            if (e.getCause() instanceof TimeoutException || e instanceof ConnectTransportException) {
-                logger.info("No response from extension to request.", e);
-            } else if (e.getCause() instanceof RuntimeException) {
-                throw (RuntimeException) e.getCause();
-            } else if (e.getCause() instanceof Error) {
-                throw (Error) e.getCause();
-            } else {
-                throw new RuntimeException(e.getCause());
+
+        logger.info("Sending extension request type: " + REQUEST_EXTENSION_ACTION_NAME);
+        transportService.getThreadPool().generic().execute(new AbstractRunnable() {
+            @Override
+            public void onFailure(Exception e) {
+                extensionIdMap.remove(extension.getId());
+                if (e.getCause() instanceof ConnectTransportException) {
+                    logger.info("No response from extension to request.", e);
+                    throw (ConnectTransportException) e.getCause();
+                } else if (e.getCause() instanceof RuntimeException) {
+                    throw (RuntimeException) e.getCause();
+                } else if (e.getCause() instanceof Error) {
+                    throw (Error) e.getCause();
+                } else {
+                    throw new RuntimeException(e.getCause());
+                }
             }
-        }
+
+            @Override
+            protected void doRun() throws Exception {
+                transportService.connectToExtensionNode(extension);
+                transportService.sendRequest(
+                    extension,
+                    REQUEST_EXTENSION_ACTION_NAME,
+                    new InitializeExtensionRequest(transportService.getLocalNode(), extension),
+                    initializeExtensionResponseHandler
+                );
+                inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
```

Comment:
As written, this will always wait for the timeout, throw an exception, and get handled in `onFailure()`.  The current implementation completes these futures in the response handler but that thread doesn't process until this one finishes.

I know we're returning immediately and this delay is on a thread that will eventually disappear without providing feedback elsewhere, but if we remove this timeout, do extensions still initialize since we're sending the transport request?  If so, recommend removing it.

Reply:
Removed the timeout.